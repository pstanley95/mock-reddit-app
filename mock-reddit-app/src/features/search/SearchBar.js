import React from "react";
import { useState, useEffect } from "react";
import styles from './resources/SearchBar.module.css'

const SearchBar = (props) => {
    const [searchValue, setSearchValue] = useState('');
    
    const handleChange = (e) => {
        setSearchValue(e.target.value);
    }
    
    const handleSubmit = (e) => {
        // do the logic
        e.preventDefault();
        setSearchValue('');
    }

    useEffect(() => {
        
    })

    return (
        <form onSubmit={handleSubmit} className={styles.form}>
            <input role="textInput" type="text" onChange={handleChange} value={searchValue} placeholder="Search for things" className={styles.textInput}/>
            <input type='button' onClick={handleSubmit} placeholder="Search" className={styles.submitButton}/>
        </form>
    )
}

export default SearchBar;