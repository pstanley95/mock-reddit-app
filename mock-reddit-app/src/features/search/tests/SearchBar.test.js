import React from 'react';
import { render, screen, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import SearchBar from '../SearchBar'
import userEvent from '@testing-library/user-event'

const mockArticles = [
    { id: 1, title: 'Article 1', content: 'Content of Article 1', initialVotes: 10},
    { id: 2, title: 'Article 2', content: 'Content of Article 2', initialVotes: -2},
    { id: 3, title: 'Article 3', content: 'Content of Article 3', initalVotes: 1000},
]

test('renders correctly', () => {
    const { asFragment } = render(<SearchBar />);
    expect(asFragment()).toMatchSnapshot();
})

test('updates value when values are typed in', () => {
    render(<SearchBar value={''} />);
    
}