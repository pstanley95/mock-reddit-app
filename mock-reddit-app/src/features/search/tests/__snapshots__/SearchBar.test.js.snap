// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <input
      class="textInput"
      placeholder="Search for things"
      role="textInput"
      type="text"
      value=""
    />
    <input
      class="submitButton"
      placeholder="Search"
      type="button"
    />
  </form>
</DocumentFragment>
`;
